gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -312,17 +315,25 @@ public synchronized void addDag(Dag<JobExecutionPlan> dag, boolean persist, bool
       log.warn("Skipping add dag because this instance of DagManager is not active for dag: {}", dag);
       return;
     }
+
+    DagId dagId = DagManagerUtils.generateDagId(dag);
     if (persist) {
-      //Persist the dag
+      // Persist the dag
       this.dagStateStore.writeCheckpoint(dag);
+      // After persisting the dag, its status will be tracked by active dagManagers so the action should be deleted
+      // to avoid duplicate executions upon leadership change
+      if (this.dagActionStore.isPresent()) {
+        this.dagActionStore.get().deleteDagAction(new DagActionStore.DagAction(dagId.getFlowGroup(),
+            dagId.getFlowName(), dagId.getFlowExecutionId(), DagActionStore.FlowActionType.LAUNCH));
```

Comment:
suggest adding a method to `DagId` (which also should be simplified as `@Data`):
```
DagAction toDagAction(FlowActionType t) {
  ...
}
```